tests/arrays/test_boolean_array.py

```diff
@@ -0,0 +1,41 @@
+from collections import UserList
+from typing import (
+    Literal,
+    cast,
+)
+
+import numpy as np
+import pandas as pd
+from pandas.core.arrays.boolean import BooleanArray
+from typing_extensions import assert_type
+
+from tests import check
+
+
+def test_construction_sequence() -> None:
+    data = cast(  # pyright: ignore[reportUnnecessaryCast]
+        "list[bool | np.bool[Literal[True]]]", [True, np.bool_(True)]
+    )
+
+    check(assert_type(pd.array(data), BooleanArray), BooleanArray)
+    check(assert_type(pd.array([*data, None]), BooleanArray), BooleanArray)
+    check(assert_type(pd.array([*data, pd.NA]), BooleanArray), BooleanArray)
+    check(assert_type(pd.array([*data, None, pd.NA]), BooleanArray), BooleanArray)
+
+    check(assert_type(pd.array(tuple(data)), BooleanArray), BooleanArray)
+    check(assert_type(pd.array(UserList(data)), BooleanArray), BooleanArray)
```

Comment:
You should add a test where the data is inline, e.g.  `pd.array([True, np.bool_(True)])`